drivers/pm_cpu_ops/pm_cpu_ops_psci.c

```diff
@@ -98,27 +104,22 @@ static int psci_features_check(unsigned long function_id)
 	return psci_data.invoke_psci_fn(PSCI_FN_NATIVE(1_0, PSCI_FEATURES), function_id, 0, 0);
 }
 
-int pm_system_reset(unsigned char reset_type)
+void z_sys_reboot(enum sys_reboot_type type)
 {
-	int ret;
-
-	if (psci_data.conduit == SMCCC_CONDUIT_NONE) {
-		return -EINVAL;
-	}
+	__ASSERT_NO_MSG(psci_data.conduit != SMCCC_CONDUIT_NONE);
 
-	if ((reset_type == SYS_WARM_RESET) &&
+	if ((type == SYS_REBOOT_WARM) &&
 	    (!psci_features_check(PSCI_FN_NATIVE(1_1, SYSTEM_RESET2)))) {
-		ret = psci_data.invoke_psci_fn(PSCI_FN_NATIVE(1_1, SYSTEM_RESET2), 0, 0, 0);
-	} else if (reset_type == SYS_COLD_RESET) {
-		ret = psci_data.invoke_psci_fn(PSCI_FN_NATIVE(0_2, SYSTEM_RESET), 0, 0, 0);
+		(void)psci_data.invoke_psci_fn(PSCI_FN_NATIVE(1_1, SYSTEM_RESET2), 0, 0, 0);
 	} else {
-		LOG_ERR("Invalid system reset type issued");
-		return -EINVAL;
+		(void)psci_data.invoke_psci_fn(PSCI_FN_NATIVE(0_2, SYSTEM_RESET), 0, 0, 0);
 	}
 
-	return psci_to_dev_err(ret);
-
+	for (;;) {
+		/* wait for reboot */
+	}
```

Comment:
I understand we cannot deal with the return value from the PSCI call anymore, but can we at least get that anyway and maybe print an error if the returned value is not the one expected?

Reply:
`sys_reboot()` has never returned a value, so we could assert or log. We could also consider allowing `sys_reboot()` to return errors, and only guarantee a no return if successful. Let's discuss later today in arch wg.

Reply:
yes, I was referring to assert or log. If then we can make `sys_reboot()` to return even better to me.

Reply:
added a printk